File: arrayConversion.cpp
int arrayConversion(vector<int> inputArray) {
    using namespace std;
    int count = 0;
    while (inputArray.size() > 1) {
        count++;
        vector<int> tmp;
        if (count % 2 == 1) {
            for (int i = 0; i < inputArray.size() / 2; i++) {
                tmp.push_back(inputArray[2 * i] + inputArray[2 * i + 1]);
            }
            inputArray = tmp;
        } else {
            for (int i = 0; i < inputArray.size() / 2; i++) {
                tmp.push_back(inputArray[2 * i] * inputArray[2 * i + 1]);
            }
            inputArray = tmp;
        }
    }
    return inputArray[0];
}

File: arrayConversion.py
def arrayConversion(inputArray):

    operation = 1
    while len(inputArray) > 1:
        newArray = []
        for i in range(0, len(inputArray), 2):
            if operation % 2 == 1:
                newArray.append(inputArray[i] + inputArray[i + 1])
            else:
                newArray.append(inputArray[i] * inputArray[i + 1])
        inputArray = newArray
        operation += 1

    return inputArray[0]
